pkgs/os-specific/linux/gt/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, lib, fetchFromGitHub, cmake, bash-completion, pkg-config, libconfig
+, asciidoc
+, libusbgx
+}:
+let repo = fetchFromGitHub {
+  owner = "linux-usb-gadgets";
+  repo = "gt";
+  rev = "7247547a14b2d092dc03fd83218ae65c2f7ff7d6";
+  sha256 = "1has9q2sghd5vyi25l3h2hd4d315vvpld076iwwsg01fx4d9vjmg";
+};
+in stdenv.mkDerivation {
+  pname = "gt";
+  version = "2021-09-30";
+  src = repo + "/source";
+  preConfigure = ''
+    cmakeFlagsArray+=("-DBASH_COMPLETION_COMPLETIONSDIR=$out/share/bash-completions/completions")
+  '';
+  nativeBuildInputs = [ cmake pkg-config asciidoc ];
+  buildInputs = [ bash-completion libconfig libusbgx];
+
+  meta = {
+    description = "Linux command line tool for setting up USB gadgets using configfs";
+    license = with lib.licenses; [ asl20 ];
+    maintainers = with lib.maintainers; [ lheckemann ];
+    platforms = lib.platforms.linux;
+  };
```

Comment:
Is this an agreed-upon standard, or your personal preference? I prefer limiting the scope of `with`s as much as possible, and would only change this if this is the standard in nixpkgs.

Reply:
> Is this an agreed-upon standard, or your personal preference?

That is the defacto standard.

> I prefer limiting the scope of withs as much as possible

That scope is really narrow. That applies to for example modules where it is often used over 100s of lines.

Reply:
```
[linus@ordnungsamd:~/nixpkgs/master]$ rg 'licenses? = (with )?lib' | wc -l
20143

[linus@ordnungsamd:~/nixpkgs/master]$ rg 'licenses? = (with )?licen' | wc -l
15400
```
doesn't look very de facto to me. If preferring `with lib;` on all of meta is _documented_ somewhere though, I'm happy to change this.

Reply:
Counting with grep does not proof anything because not everything is updated immediately.  For example the pname+version conversion is ongoing for years.